docs/constants/navigation.js

```diff
@@ -1,574 +1,582 @@
-
 module.exports = {
-  generalDirectories: prevaledNavigationData.generalDirectories,
-  startingDirectories: prevaledNavigationData.startingDirectories,
-  previewDirectories: prevaledNavigationData.previewDirectories,
-  easDirectories: prevaledNavigationData.easDirectories,
-  featurePreviewDirectories: prevaledNavigationData.featurePreviewDirectories,
-  starting: sortedStarting,
-  general: sortedGeneral,
-  eas: sortedEas,
-  preview: sortedPreview,
-  featurePreview: sortedFeaturePreview,
-  reference: { ...sortedReference, latest: sortedReference['v' + packageVersion] },
+  starting,
+  general,
+  eas,
+  preview,
+  featurePreview,
+  /** @type {any} */
+  reference: { ...reference, latest: reference[LATEST_VERSION] },
+  generalDirectories,
+  startingDirectories,
+  previewDirectories,
+  easDirectories,
+  featurePreviewDirectories,
   hiddenSections,
   collapsedSections,
 };
+
+// --- MDX methods ---
+
+function makeSection(name, children = []) {
+  // TODO(cedric): refactor node types to match unist
+  return { name, children };
+}
+
+function makeGroup(name, children = [], href = '') {
+  // TODO(cedric): refactor node types to match unist
+  return { name, href, posts: children };
+}
+
+/**
+ * Parse the MDX page and extract the frontmatter/yaml page information.
+ * It will only look for the frontmatter/yaml block in the root nodes.
+ * This requires the `remark-frontmatter` MDX plugin.
+ *
+ * @param {string} file
+ */
+function makePage(file) {
+  const filePath = !path.isAbsolute(file) ? path.resolve(PAGES_DIR, file) : file;
+  const contents = fs.readFileSync(filePath, 'utf-8');
+  const url = pageUrl(filePath);
+  const data = frontmatter(contents).attributes;
+
+  if (!data) {
+    console.error('Page YAML block is unreadable:', file);
+  } else if (!data.title) {
+    console.error('Page does not have a `title`:', file);
+    data.title = '';
+  }
+
+  const result = {
+    // TODO(cedric): refactor name into title
+    name: data.title,
+    // TODO(cedric): refactor href into url
+    href: url,
+  };
+  // TODO(cedric): refactor sidebarTitle into metadata
+  if (data.sidebar_title) {
+    result.sidebarTitle = data.sidebar_title;
+  }
+  // TODO(cedric): refactor sidebarTitle into metadata
+  if (data.hidden) {
+    result.hidden = data.hidden;
+  }
+  return result;
+}
+
+// --- Other helpers ---
+
+/**
+ * Load all pages from a single directory.
+ */
+function pagesFromDir(dir) {
+  return fs
+    .readdirSync(path.resolve(PAGES_DIR, dir), { withFileTypes: true })
+    .filter(entity => entity.isFile())
+    .map(file => makePage(path.join(dir, file.name)));
+}
+
+/**
+ * Create the page url using the absolute file path.
+ * This parses the URL, relatively from PAGES_DIR.
+ * It also strips the file extension, and name if its `index`.
+ */
+function pageUrl(file) {
+  const filePath = path.parse(file);
+  const { pathname } = new URL(path.relative(PAGES_DIR, file), 'https://docs.expo.dev');
```

Comment:
Swapped out the previous hack and slash dirty URL from path generation with these things. The URL should always return a valid pathname, starting with a `/`. This seems way safer compared to relying on the paths, which is different per OS.